hw/xbox/nv2a/nv2a_psh.c

```diff
@@ -434,28 +456,27 @@ static void add_stage_code(struct PixelShader *ps,
 
     QString *ab_mapping = get_output(ab, output.mapping);
     QString *cd_mapping = get_output(cd, output.mapping);
-    QString *ab_dest = get_var(ps, output.ab, true);
-    QString *cd_dest = get_var(ps, output.cd, true);
-    QString *sum_dest = get_var(ps, output.muxsum, true);
+    QString *ab_dest = get_var(ps, output.ab);
+    QString *cd_dest = get_var(ps, output.cd);
+    QString *sum_dest = get_var(ps, output.muxsum);
 
-    if (qstring_get_length(ab_dest)) {
-        qstring_append_fmt(ps->code, "%s.%s = %s(%s);\n",
-                           qstring_get_str(ab_dest), write_mask, caster, qstring_get_str(ab_mapping));
+    if (ab_dest != NULL) {
+        qstring_append_fmt(ps->code, "%s.%s = clamp(%s, -1.0, 1.0);\n",
+                           qstring_get_str(ab_dest), write_mask, qstring_get_str(ab_mapping));
```

Comment:
The `} else {` which follows makes no sense: uses RHS as LHS?! This code needs to be rewritten